### PR TITLE
Feature/farm 62 view prof loss statement

### PR DESCRIPTION
**Pull request by isaacmartinezimv**
[[FARM-62]-View P/L statement](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-62)

#### 📲 What
Profit and Loss screen View
#### 🤔 Why
		
Because farmers need to see their transactions in order to have a balance of theirs sales and cost		
#### 🛠 How

New UI design with new widgets
Externalised widgets
Local style/theme variables
TODO: update typography
OUT OF SCOPE: All, Cost, Sales tabs; day, week feature

#### 👀 See
		
gif_ProffitAndLoss2
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [X] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [X] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [X] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [X] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [X] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

### CODE REVIEW

File: lib/ui/common/CompactRoundedBackgroundTextStyle.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
+import 'RoundedBackgroundText.dart';
+
+class PositiveCompactRoundedBackgroundTextStyle implements RoundedBackgroundTextStyle {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/CompactRoundedBackgroundTextStyle.dart]: How can we make this name shorter?
How about renaming RoundedBackgroundText to Dogtag ?
PositiveDogTag , NegativeDogTag

The presence on compact in the name make me think we have variants, do we have large and small styles? if not drop the compact.

File: lib/ui/common/RoundedBackgroundText.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+class RoundedBackgroundTextViewModel {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/RoundedBackgroundText.dart]: See above with name change (maybe call this DogTag)

File: lib/ui/common/RoundedBackgroundText.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  String title;
+  IconData icon;
+
+  RoundedBackgroundTextViewModel({this.title, this.icon});
+}
+
+RoundedBackgroundTextViewModel buildBackgroundTextViewModel({String title, IconData icon}) {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/RoundedBackgroundText.dart]: not sure of the point of this?

File: lib/ui/common/RoundedBackgroundText.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+class RoundedBackgroundText {
+  static build(
+  {RoundedBackgroundTextStyle style = const _DefaultStyle(),
+  String title, IconData icon}) {
+    return _buildRoundedBackgroundText(style, title: title, icon: icon);
+  }
+
+  static Widget _buildRoundedBackgroundText(RoundedBackgroundTextStyle style, {String title, icon}) {
+    String defaultValue = "0";
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/RoundedBackgroundText.dart]: what's default value? this function should pass in the viewModel, mock the viewModel outside the widget for testing. Davids demo widget will help with this.

File: lib/ui/common/RoundedBackgroundText.dart [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    RoundedBackgroundTextViewModel viewModel = buildBackgroundTextViewModel(title: title, icon: icon);
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/RoundedBackgroundText.dart]: pass this in.

File: lib/ui/profitloss/ProfitLossList.dart [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+class ProfitLossViewModel {
+  final String title;
+  final String titleDetail;
+  final String subTitle;
+
+  ProfitLossViewModel(this.title, this.titleDetail, this.subTitle);
+}
+
+ProfitLossViewModel buildProfitLossViewModel() {
+  return ProfitLossViewModel("2,150", "KSh", "▲ 498 (17.4%)");
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: always Mock outside, so you don't have to come back and change the widget code.

File: lib/ui/profitloss/ProfitLossList.dart [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+abstract class ProfitLossStyle {
+
+  final TextStyle titleTextStyle;
+  final TextStyle titleDetailTextStyle;
+  final TextStyle subTitleTextStyle;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: typo should be "subtitle"

File: lib/ui/profitloss/ProfitLossList.dart [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+  final Color floatingButtonBackgroundColor;
+  final EdgeInsets generalMargins;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: what's general margins?

File: lib/ui/profitloss/ProfitLossList.dart [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+  return ProfitLossViewModel("2,150", "KSh", "▲ 498 (17.4%)");
+}
+
+abstract class ProfitLossStyle {
+
+  final TextStyle titleTextStyle;
+  final TextStyle titleDetailTextStyle;
+  final TextStyle subTitleTextStyle;
+
+  final Color floatingButtonBackgroundColor;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: try to avoid a name that hint at the implementation specifics. Maybe name this actionButtonBackgroundColour.
maybe if won't float in the future, but we´ll always have the button to perform the P and L action.

File: lib/ui/profitloss/ProfitLossListItem.dart [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+  ProfitLossItemStyle(this.elevation, this.edgePadding, this.detailLineSpace, this.titleStyle, this.maxLines,
+      this.titleLineSpace, this.subTitleStyle);
+
+}
+
+class _DefaultStyle implements ProfitLossItemStyle {
+
+  final TextStyle titleStyle = const TextStyle(fontSize: 12, fontWeight: FontWeight.normal, color: Color(0xff767690));
+  final TextStyle subTitleStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.normal, color: Color(0xff1a1b46));
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossListItem.dart]: typo should be subtitle

File: lib/ui/profitloss/ProfitLossListItem.dart [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+  final double elevation = 0.0;
+  final double detailLineSpace = 20.5;
+  final int maxLines = 1;
+  final double titleLineSpace = 7;
+
+  const _DefaultStyle();
+}
+
+class HomeProfitLossChild {
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossListItem.dart]: naming? is the view model named the same? what is this? a list? a listItem? 
All lists are named consistently. <object we are listing>List ...ListItem ..ViewModel ...Style
(eg. in our last PR review: plotList, plotListStyle, plotListViewModel) plotListItem, plotListItemViewModel etc etc.....)

File: lib/ui/profitloss/profit_loss_viewmodel.dart [hunk 12]
```diff
@@ -0,0 +1,14 @@
+
+import 'package:farmsmart_flutter/model/loading_status.dart';
+import 'package:farmsmart_flutter/redux/app/app_state.dart';
+import 'package:redux/redux.dart';
+
+class MyProfitLossViewModel {
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/profit_loss_viewmodel.dart]: drop "My"

File: lib/ui/profitloss/ProfitLossList.dart [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+  final double floatingButtonElevation;
+  final double floatingButtonIconSize;
+
+  ProfitLossStyle(this.floatingButtonBackgroundColor, this.floatingButtonSize,
+      this.floatingButtonElevation, this.floatingButtonIconSize, this.generalMargins,
+      this.titleTextStyle, this.titleDetailEdgePadding, this.titleDetailTextStyle,
+      this.subTitleTextStyle);
+}
+
+class _DefaultProfitLossStyle implements ProfitLossStyle{
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: we have the context already. all default styles should be named _DefaultStyle

File: lib/ui/profitloss/ProfitLossList.dart [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+abstract class ProfitLossStyle {
+
+  final TextStyle titleTextStyle;
+  final TextStyle titleDetailTextStyle;
+  final TextStyle subTitleTextStyle;
+
+  final Color floatingButtonBackgroundColor;
+  final EdgeInsets generalMargins;
+
+  final double titleDetailEdgePadding;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: titleDetail???
maybe should be detailText?

File: lib/ui/profitloss/ProfitLossList.dart [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+import 'package:farmsmart_flutter/ui/profitloss/profit_loss_viewmodel.dart';
+import 'package:farmsmart_flutter/utils/strings.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:farmsmart_flutter/ui/common/headerAndFooterListView.dart';
+
+class ProfitLossViewModel {
+  final String title;
+  final String titleDetail;
+  final String subTitle;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: subtitle

File: lib/ui/profitloss/ProfitLossList.dart [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+import 'package:farmsmart_flutter/ui/profitloss/ProfitLossListItem.dart';
+import 'package:farmsmart_flutter/ui/profitloss/profit_loss_viewmodel.dart';
+import 'package:farmsmart_flutter/utils/strings.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:farmsmart_flutter/ui/common/headerAndFooterListView.dart';
+
+class ProfitLossViewModel {
+  final String title;
+  final String titleDetail;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: detailText

File: lib/ui/profitloss/ProfitLossList.dart [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+      case LoadingStatus.ERROR:
+        return Text(Strings.errorString);
+    }
+  }
+
+  Widget _buildPage(BuildContext context, MyProfitLossViewModel viewModel, ProfitLossViewModel profitLossViewModel, ProfitLossStyle profitStyle) {
+    final viewModel = buildProfitLossViewModel();
+    return HeaderAndFooterListView.builder(
+      //FIXME: itemCount is now hardcoded
+        itemCount: 10,
```
Comment (LeeMatthewHiggins) [inline lib/ui/profitloss/ProfitLossList.dart]: Use the ViewModel and inject a mock. don't add any test code to the widgets. This is the point of the ViewModel.
We want to be able to sign off the widget in this PR, and make no changes when we hook it up to real data.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]
Good start.
Please make sure naming is consistent with our previous reviews.

please use Mock view models and inject them to test your widget.

File: lib/ui/common/ActionSheet.dart [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+        actionItemHeight: actionItemHeight ?? this.actionItemHeight,
+        iconLineSpace: iconLineSpace ?? this.iconLineSpace,
+        actionItemElevation: actionItemElevation ?? this.actionItemElevation,
+        iconHeight: iconHeight ?? this.iconHeight,
+        maxLines: maxLines ?? this.maxLines
+    );
+  }
+}
+
+class ActionSheet extends StatelessWidget{
```
Comment (davidfranquet) [inline lib/ui/common/ActionSheet.dart:117]: As talked, we shouldn't rebase/merge from another branch that is not yours and is not merged to develop

File: lib/ui/common/Dogtag.dart [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+                  children: _buildButtonContent(),
+                  spacing: style.spacing,
+                ))
+          ],
+        ),
+      );
+    }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (davidfranquet) [inline lib/ui/common/Dogtag.dart]: @override functions should be before private and public functions

File: lib/ui/common/Dogtag.dart [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+  }
+}
+
+class DogTag extends StatelessWidget {
+  final DogTagViewModel _viewModel;
+  final DogTagStyle _style;
+
+  const DogTag({Key key, DogTagViewModel viewModel, DogTagStyle style }) : this._viewModel = viewModel, this._style = style, super(key:key);
+
+  static Widget _build(DogTagViewModel viewModel, DogTagStyle style) {
```
Comment (davidfranquet) [inline lib/ui/common/Dogtag.dart]: Don't think that this function should be here. You can move all this stuff inside the @override build and remove this function

File: lib/ui/common/Dogtag.dart [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+
+    List<Widget> _buildButtonContent(){
+      List<Widget> listBuilder = [];
+
+      if (viewModel.icon!= null) {
```
Comment (davidfranquet) [inline lib/ui/common/Dogtag.dart]: Missing space
```suggestion
      if (viewModel.icon != null) {
```

File: lib/ui/common/Dogtag.dart (hunk 20 above)
Comment (davidfranquet) [reply lib/ui/common/Dogtag.dart]: If you have any doubts ask me :)

File: lib/ui/common/roundedButton.dart [hunk 22]
```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
...
-        SnackBar(
-          content: Text(
-              toastText
-          ),
-          action: SnackBarAction(label: toastButtonText, onPressed: scaffold.hideCurrentSnackBar),
-        )
-    );
+  @override
+  Widget build(BuildContext context) {
+    return _build(_viewModel, _style);
```
Comment (davidfranquet) [inline lib/ui/common/roundedButton.dart]: Move all the stuff inside _build here and remove the _build function. Remember that @override funcs should be before private and public funcs

File: lib/ui/discover/ArticleListItemStyle.dart [hunk 23]
```diff
@@ -17,6 +17,8 @@ abstract class ArticleListItemStyle {
   final int maxLinesPerTitle;
   final int maxLinesPerSummary;
 
+
```
Comment (davidfranquet) [inline lib/ui/discover/ArticleListItemStyle.dart]: Unnecessary CR

File: lib/ui/discover/HeroListItem.dart [hunk 24]
```diff
@@ -32,8 +32,7 @@ class _DefaultStyle implements ArticleListItemStyle {
 }
 
 class HeroListItem {
```
Comment (davidfranquet) [inline lib/ui/discover/HeroListItem.dart:34]: Covert to StatelessWidget

File: lib/ui/home.dart [hunk 25]
```diff
@@ -49,10 +55,15 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
 
   final List<Widget> _children = [
     PlotList(),
-    HomeProfitLossChild(),
+    ProfitLossPage(),
     ArticleList(),
     HomeCommunityChild(),
-    PlaygroundView(widgetList: [], appBarColor: Color(0xFF9CBD3A),),
+    PlaygroundView(widgetList: [
```
Comment (davidfranquet) [inline lib/ui/home.dart]: @jessicahernandezhv moved it to a separated file. Do the same with the same file name (you will have conflicts)

File: lib/ui/myplot/PlotList.dart [hunk 26]
```diff
@@ -76,76 +75,103 @@ class _MyPlotState extends State<PlotList> {
...
+        children: <Widget>[
+          AlertDialog(
+              title: Text(
+                  plotStyle.errorText),
+              actions: <Widget>[
+                FlatButton(
+                    child: Text(
+                        retryButton),
+                    onPressed: () {
+                      //FIXME: Needs to implement the retry functionality
```
Comment (davidfranquet) [inline lib/ui/myplot/PlotList.dart]: this is out of scope?

File: lib/ui/profitloss/ProfitLossList.dart [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+    );
+    }
+
+    Widget _buildSubTitle(ProfitLossListViewModel viewModel, ProfitLossStyle profitStyle) {
+    return Container(
+      child: GestureDetector(
+        child: Text(
+          viewModel.subtitle,
+          style: profitStyle.subtitleTextStyle
+    ),
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossList.dart]: Fix TAB

File: lib/ui/profitloss/ProfitLossList.dart [hunk 28]
```diff
@@ -0,0 +1,195 @@
...
+        onTap: () {
+          //FIXME: Add navigation to the next screen when finished
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossList.dart]: out of scope?

File: lib/ui/profitloss/ProfitLossListItem.dart [hunk 29]
```diff
@@ -0,0 +1,93 @@
...
+
+  final double elevation = 0.0;
+  final double detailLineSpace = 20.5;
+  final int maxLines = 1;
+  final double titleLineSpace = 7;
+
+  const _DefaultStyle();
+}
+
+class ProfitLossListItem {
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossListItem.dart]: Statelesswidget :)

File: lib/ui/profitloss/mockRepositoryTryout/MockTransactionRepository.dart [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+      detailText: _mockDetailText.random(),
+      subtitle: _mockDetailText.random(),
+      transactions: list
+    );
+  }
+}
+
+MockString _mockTitleText = MockString(library: [
+  "Title example",
+"Longer title example ",
```
Comment (davidfranquet) [inline lib/ui/profitloss/mockRepositoryTryout/MockTransactionRepository.dart]: Check your TAB settings, something wrong with your tabs

**Review by davidfranquet — CHANGES_REQUESTED** [review]
As talked we shouldn't rebase our branch with any branch that isn't your 'parent branch'. Also check my comments

File: lib/ui/common/Dogtag.dart (hunk 20 above)
Comment (isaacmartinezimv) [reply lib/ui/common/Dogtag.dart]: Thanks for the help !

File: lib/ui/home.dart [hunk 31]
```diff
@@ -80,6 +102,7 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
       bottomNavigationBar: Theme(
         data: Theme.of(context).copyWith(primaryColor: Color(primaryGreen)),
         child: BottomNavigationBar(
+
```
Comment (davidfranquet) [inline lib/ui/home.dart]: remove unnecessary CR

File: lib/ui/profitloss/ProfitLossHeader.dart [hunk 32]
```diff
@@ -0,0 +1,111 @@
...
+  }
+}
+
+class ProfitLossHeader extends StatelessWidget {
+  final ProfitLossHeaderViewModel _viewModel;
+  final ProfitLossHeaderStyle _style;
+
+  const ProfitLossHeader({Key key, ProfitLossHeaderViewModel viewModel, ProfitLossHeaderStyle style}) : this._viewModel = viewModel, this._style = style, super(key: key);
+
+  static Widget _build(BuildContext context, ProfitLossHeaderViewModel viewModel, ProfitLossHeaderStyle style) {
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossHeader.dart]: Do the same as DogTag widget

File: lib/ui/profitloss/ProfitLossList.dart [hunk 33]
```diff
@@ -0,0 +1,116 @@
...
+  final String detailText;
+
+  final List<ProfitLossListItemViewModel> transactions;
+
+  ProfitLossListViewModel({this.title, this.detailText, this.loadingStatus, this.transactions});
+
+  static ProfitLossListViewModel fromStore(Store<AppState> store) {
+    final mockViewModel = MockProfitLossListViewModel.build();
+
+    return mockViewModel;
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossList.dart]: Join the return with the instance 
```suggestion
    return MockProfitLossListViewModel.build();;
```

File: lib/ui/profitloss/ProfitLossHeader.dart [hunk 34]
```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/material.dart';
+
+class ProfitLossHeaderViewModel {
+  String title;
+  String detail;
+
+  ProfitLossHeaderViewModel(this.title, this.detail);
+}
+
+class ProfitLossHeaderStyle {
```
Comment (davidfranquet) [inline lib/ui/profitloss/ProfitLossHeader.dart:10]: Add missing commas and do a format code :)

File: lib/ui/myplot/PlotList.dart (hunk 26 above)
Comment (LeeMatthewHiggins) [reply lib/ui/myplot/PlotList.dart]: This should call a ViewModel function. Suggest a var in the view model. ....  Function update;
The Bloc will then fill this in later.

File: lib/ui/common/ActionSheet.dart [hunk 35]
```diff
@@ -0,0 +1,236 @@
...
+
+  static Widget _buildActionCell(ActionSheetStyle style, ActionListItemViewModel viewModel,
+      int numberOfActions, int currentAction) {
+    return Column(
+      children: <Widget>[
+        Card(
+          elevation: style.actionItemElevation,
+          color: style.actionItemBackgroundColor,
+          child: InkWell(
+            onTap: () => print(viewModel.title), //No mock here -> View Model
```
Comment (albertmiro) [inline lib/ui/common/ActionSheet.dart:168]: We should have this print here? Or is missing the call to the viewmodel?

File: lib/ui/common/ActionSheet.dart [hunk 36]
```diff
@@ -0,0 +1,236 @@
...
+      Image.asset(viewModel.checkBoxIcon, height: style.iconHeight)
+        ],
+      ));
+    }
+    return listBuilder;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _build(context, _viewModel, _style);
```
Comment (albertmiro) [inline lib/ui/common/ActionSheet.dart:226]: Instead of having this private build, let's have here what's inside _build.

File: lib/ui/common/Dogtag.dart [hunk 37]
```diff
@@ -0,0 +1,159 @@
...
+            fontSize: 15,
+            fontWeight: FontWeight.normal,
+            color: Color(0xff25df0c)),
+        maxLines: 1,
+        iconSize: 8,
+        spacing: 5.5);
+  }
+
+  factory DogTagStyle.positiveStyle() {
+    return DogTagStyle.defaultStyle().copyWith();
```
Comment (albertmiro) [inline lib/ui/common/Dogtag.dart]: Is needed the copyWith if we are not changing any parameter?

File: lib/ui/common/roundedButton.dart [hunk 38]
```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
...
+        height: 24.0,
+        width: 24.0,
+        iconEdgePadding: 0,
+        buttonIconSize: 9.0
+    );
+  }
 
-class _DefaultStyle implements RoundedButtonStyle {
+  factory RoundedButtonStyle.compactRoundedButton() {
+    return RoundedButtonStyle.defaultStyle().copyWith();
```
Comment (albertmiro) [inline lib/ui/common/roundedButton.dart]: Same as above

File: lib/ui/myplot/PlotList.dart [hunk 39]
```diff
@@ -76,76 +74,101 @@ class _MyPlotState extends State<PlotList> {
...
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  children: <Widget>[
+                    Text(
+                      viewModel.title,
+                      style: myPlotStyle.titleTextStyle,
+                    )
+                  ]
+              ),
+              RoundedButton(viewModel:
+                  RoundedButtonViewModel(icon: "assets/icons/profit_add.png", onTap: () => _showToast(context)),
```
Comment (albertmiro) [inline lib/ui/myplot/PlotList.dart]: Don't think we should have this hardcoded text here with the asset name, move it to a constant

File: lib/ui/profitloss/ProfitLossHeader.dart [hunk 40]
```diff
@@ -0,0 +1,124 @@
...
+          child: Text(
+              viewModel.subtitle,
+              style: style.subtitleTextStyle
+          ),
+          onTap: () {
+            //FIXME: Add navigation to the next screen when finished
+          },
+        ));
+  }
+*/
```
Comment (albertmiro) [inline lib/ui/profitloss/ProfitLossHeader.dart:123]: If that should be done in the future is ok, if it's not going to be used, let's remove the comment

File: lib/ui/profitloss/ProfitLossList.dart [hunk 41]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  Widget _buildPageWithFloatingButton(BuildContext context,
+      ProfitLossListViewModel viewModel, ProfitLossStyle profitStyle) {
+    return Scaffold(
+        floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
+        body: _buildPage(context, viewModel, profitStyle),
+        floatingActionButton: RoundedButton(
+            viewModel: RoundedButtonViewModel(
+                icon: "assets/icons/profit_add.png",
```
Comment (albertmiro) [inline lib/ui/profitloss/ProfitLossList.dart]: Move this text to a constant.

File: lib/ui/profitloss/ProfitLossListItem.dart [hunk 42]
```diff
@@ -0,0 +1,124 @@
...
+      ProfitLossListItemViewModel viewModel,
+      ProfitLossItemStyle style})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+        //onTap: viewModel.onTap,
```
Comment (albertmiro) [inline lib/ui/profitloss/ProfitLossListItem.dart]: If not used, remove the comment

File: lib/ui/common/Dogtag.dart [hunk 43]
```diff
@@ -0,0 +1,159 @@
...
+  }
+
+  factory DogTagStyle.negativeStyle() {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/Dogtag.dart:48]: don't have specific use cases in a generic widget. (negative and positive style makes no sense generically)
Just have them defined were they are used or if shared in their own file.

File: lib/ui/common/Dogtag.dart [hunk 44]
```diff
@@ -0,0 +1,159 @@
...
+        iconSize: iconSize ?? this.iconSize,
+        spacing: spacing ?? this.spacing);
+  }
+}
+
+class DogTag extends StatelessWidget {
+  final DogTagViewModel _viewModel;
+  final DogTagStyle _style;
+
+  const DogTag({Key key, DogTagViewModel viewModel, DogTagStyle style})
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/Dogtag.dart:90]: this does not follow the same pattern as other PR´s why?
There should be a default set.

File: lib/ui/common/roundedButton.dart [hunk 45]
```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
...
+  }
 
-  final Color iconButtonColor =  const Color(0xFFFFFFFF);
-  final Color backgroundColor =  const Color(0xff24d900);
+  factory RoundedButtonStyle.compactBigRoundedButton() {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/roundedButton.dart]: compactBig.  ????!!!

File: lib/ui/common/roundedButton.dart [hunk 46]
```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
...
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+    );
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.actionSheetLargeRoundedButton(){
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/roundedButton.dart]: why does a generic component now know about action sheets? Don't add every style to a generic component! imagine how big this file would get.  It should just have a default style, that's it! 
The other styles should be created at the point of use.... so action sheet should own this.

File: lib/ui/common/roundedButton.dart [hunk 47]
```diff
@@ -2,125 +2,152 @@ import 'package:flutter/material.dart';
...
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: <Widget>[
+          Padding(
+            padding: _style.edgePadding,
+            child: Container(
+              alignment: Alignment.center,
+              height: _style.height,
+              width: _style.width,
+              decoration: BoxDecoration(
+                  color: _viewModel.backgroundColor != null ? _viewModel.backgroundColor : _style.backgroundColor,
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/roundedButton.dart]: why viewmodel involved here? that's not the place for colours . it can have things like "isSelected" and then the style has selected and non selected styles. But it shouldn't drive the style like this.

File: lib/ui/discover/discover_page.dart [hunk 48]
```diff
@@ -85,7 +85,7 @@ class _DiscoveryState extends State<ArticleList> {
         itemBuilder: (BuildContext context, int index) {
           final viewModel = fromArticleEntityToViewModel(articlesList[index], getRelatedArticles);
           if (index == 0) {
```
Comment (LeeMatthewHiggins) [inline lib/ui/discover/discover_page.dart:87]: don't change any of the discovery section files. this is being refactored.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]
Please revert all changes to anything in the discover sections this is being refactored and is not related to this PR.

Please address other comments.

File: lib/ui/profitloss/ProfitLossListItem.dart (hunk 42 above)
Comment (LeeMatthewHiggins) [reply lib/ui/profitloss/ProfitLossListItem.dart]: it should be used.....

File: lib/ui/common/roundedButton.dart (hunk 46 above)
Comment (isaacmartinezimv) [reply lib/ui/common/roundedButton.dart]: This will be changed in the Jessica PR, because is related with the ActionSheet

File: lib/ui/profitloss/ProfitLossList.dart (hunk 41 above)
Comment (isaacmartinezimv) [reply lib/ui/profitloss/ProfitLossList.dart]: Thanks for the explanation Albert !

File: lib/ui/common/roundedButton.dart [hunk 49]
```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
...
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
+  RoundedButtonStyle({
+    this.height,
+    this.width,
+    this.iconButtonColor,
+    this.backgroundColor,
+    this.iconEdgePadding,
+    this.buttonShape,
+    this.borderRadius,
+    this.edgePadding,
```
Comment (davidfranquet) [inline lib/ui/common/roundedButton.dart:31]: I think that we shouldn't have the padding inside the widget, the padding should be set on the parent widget that contain this parent. Because If I want to to use this widget I'll have this default padding and maybe I want to have less padding.

File: lib/ui/common/Dogtag.dart (hunk 43 above)
Comment (isaacmartinezimv) [reply lib/ui/common/Dogtag.dart:48]: The styles will be refactored in a coming PR

File: lib/ui/common/Dogtag.dart (hunk 44 above)
Comment (isaacmartinezimv) [reply lib/ui/common/Dogtag.dart:90]: The styles will be refactored for the new format in the next's PR

**Comment by isaacmartinezimv** [comment]
> Please revert all changes to anything in the discover sections this is being refactored and is not related to this PR.
> 
> Please address other comments.

reverted all changes on discovery section

File: lib/ui/common/roundedButton.dart [hunk 50]
```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
...
+      width: 24.0,
+      iconEdgePadding: 0,
+      buttonIconSize: 9.0,
+    );
+  }
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+  factory RoundedButtonStyle.bigRoundedButton() {
```
Comment (davidfranquet) [inline lib/ui/common/roundedButton.dart:52]: Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one

File: lib/ui/common/roundedButton.dart [hunk 51]
```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
...
+    return RoundedButtonStyle.defaultStyle().copyWith(
+        borderRadius: BorderRadius.all(Radius.circular(40)),
+        height: 48.0,
+        width: 48.0,
+        buttonIconSize: 14.0);
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.largeRoundedButtonStyle() {
```
Comment (davidfranquet) [inline lib/ui/common/roundedButton.dart:61]: Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one

File: lib/ui/common/roundedButton.dart [hunk 52]
```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
...
+  }
 
-class RoundedButton {
-  static build(
-      {RoundedButtonStyle style = const _DefaultStyle(),
-        @required  BuildContext context,
-        String title, IconData icon,
-        Function onTap}) {
-    return _buildRoundedButton(style, context , title: title, icon: icon, onTap: onTap);
+  factory RoundedButtonStyle.actionSheetLargeRoundedButton() {
```
Comment (davidfranquet) [inline lib/ui/common/roundedButton.dart:75]: Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one

File: lib/ui/playground/data/playground_tasks_datasource.dart [hunk 53]
```diff
@@ -8,9 +10,17 @@ class PlayGroundTasksDataSource implements PlaygroundDataSource {
...
-        title: 'TASK FARM-402 Test playground widget',
-        child: Text('Test play ground'),
+        title: 'TASK FARM-62 View-Prof-Loss-Statement Header',
+        child: ProfitLossHeader(viewModel: MockProfitLossHeaderViewModel.build(), style: ProfitLossHeaderStyle.defaultStyle())
       ),
+      PlaygroundWidget(
+        title: 'TASK FARM-62 View-Prof-Loss-Statement Negative',
+        child: ProfitLossListItem(viewModel: MockProfitLossListItemViewModel.buildNegative(), style: ProfitLossItemStyle.defaultStyle()),
+      ),
+    PlaygroundWidget(
```
Comment (davidfranquet) [inline lib/ui/playground/data/playground_tasks_datasource.dart]: Format code.......

Comment (davidfranquet) [reply lib/ui/playground/data/playground_tasks_datasource.dart]: Option + Cmd + L

**Comment by davidfranquet** [comment]
@isaacmartinezimv - Also you have a empty file: https://github.com/farmsmart/farmsmart-flutter/blob/e78170d04e5830a505bfa48b37a7a921235325c4/lib/ui/common/PlaygroundWidgetTryOut.dart

Remove it if you don't need it

**Review by LeeMatthewHiggins — APPROVED** [review]

**Review by davidfranquet — APPROVED** [review]
